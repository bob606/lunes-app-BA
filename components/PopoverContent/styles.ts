import {StyleSheet} from 'react-native';
import {widthPercentageToDP as wp} from 'react-native-responsive-screen';
import {COLORS} from '../../constants/colors';

export const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: COLORS.lunesBlack,
    width: wp('80%'),
    height: 60,
    paddingVertical: 8,
    paddingHorizontal: 9,
    borderRadius: 2,
  },
  message: {
    color: COLORS.lunesWhite,
    fontSize: wp('3.5%'),
    fontWeight: 'normal',
    fontFamily: 'SourceSansPro-Regular',
    width: wp('60%'),
    marginLeft: 8,
  },
});
