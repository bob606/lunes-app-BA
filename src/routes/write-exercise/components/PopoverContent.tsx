import React, { ReactElement } from 'react'
import { InfoIcon } from '../../../../assets/images'
import { widthPercentageToDP as wp } from 'react-native-responsive-screen'
import labels from '../../../constants/labels.json'
import styled from 'styled-components/native'

const StyledContainer = styled.View`
  flex-direction: row;
  align-items: center;
  background-color: ${prop => prop.theme.colors.lunesBlack};
  width: ${wp('80%')}px;
  height: 60px;
  padding: 9px 8px 9px 8px;
  border-radius: 2px;
`
const StyledMessage = styled.Text`
  color: ${prop => prop.theme.colors.lunesWhite};
  font-size: ${wp('3.5%')}px;
  font-weight: ${props => props.theme.fonts.lightFontWeight};
  font-family: ${props => props.theme.fonts.contentFontRegular};
  width: ${wp('60%')}px;
  margin-left: 8px;
`

const PopoverContent = (): ReactElement => (
  <StyledContainer>
    <InfoIcon width={30} height={30} />
    <StyledMessage>{labels.exercises.write.feedback.articleMissing}</StyledMessage>
  </StyledContainer>
)

export default PopoverContent
