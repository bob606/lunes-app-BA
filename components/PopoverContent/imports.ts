import React from 'react';
import {View, Text} from 'react-native';
import {styles} from './styles';
import {InfoIcon} from '../../assets/images/imports';

export {React, View, Text, styles, InfoIcon};
