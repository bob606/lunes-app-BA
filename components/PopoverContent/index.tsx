import {Text, React, styles, View, InfoIcon} from './imports';

const PopoverContent = () => (
  <View style={styles.container}>
    <InfoIcon width={30} height={30} />
    <Text style={styles.message}>
      Ups, you forgot someting. You have to add an article to continue.
    </Text>
  </View>
);

export default PopoverContent;
