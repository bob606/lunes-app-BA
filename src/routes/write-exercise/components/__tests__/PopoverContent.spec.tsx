import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'
import 'react-native'

import PopoverContent from '../PopoverContent'

describe('components', () => {
  describe('PopoverContent', () => {
    it('renders correctly across screens', () => {
      const component = shallow(<PopoverContent />)
      expect(toJson(component)).toMatchSnapshot()
    })
  })
})
