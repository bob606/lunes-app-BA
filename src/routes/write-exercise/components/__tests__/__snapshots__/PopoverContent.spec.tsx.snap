// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components PopoverContent renders correctly across screens 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000e38",
      "borderRadius": 2,
      "flexDirection": "row",
      "height": 60,
      "paddingHorizontal": 9,
      "paddingVertical": 8,
      "width": 600,
    }
  }
>
  <SvgrURL
    height={30}
    width={30}
  />
  <Text
    style={
      Object {
        "color": "#f9fafb",
        "fontFamily": "SourceSansPro-Regular",
        "fontSize": 26.5,
        "fontWeight": "normal",
        "marginLeft": 8,
        "width": 450,
      }
    }
  >
    Ups, deine Eingabe ist nicht vollständig. Füge einen Artikel ein.
  </Text>
</View>
`;
